MariaDB/server
# MDEV-28628: Change current Debian package revision scheme

- [x] *The Jira issue number for this PR is: MDEV-28628*

## Description
Current Debian package revision scheme when using debian/autobake-deb.sh script is:
  `1:VERSION+maria~LSBNAME`

For example if VERSION can be like 10.6.8 and LSBNAME is buster then version and revision is:
  `1:10.6.8+maria~buster`
Which can lead to problem as distro code names can be lexical unordered.

For example Debian LSBNAME's can be:
  Codename Buster is Debian version 10
  Codename Bookworm is Debian version 11

This happens because in ASCII table Buster first two digits are 'Bu' and they are in hex 0x42 and _0x75_ and Bookworm first digits 'Bo' are they are in hex 0x42 and _0x6F_. When apt is upgrading it  means that:
  `1:10.6.8+maria~buster` is bigger than `1:10.6.8+maria~bookworm`
and that leads to problems in dist-upgrade process

To solve problem revision format is changed to:
  `1:VERSION+maria~(deb|ubu)LSBVERSION`

Example for Debian 11 is now:
  `1:10.6.8+maria~deb11`

and for Ubuntu 22.04 is now:
  `1:10.6.8+maria~ubu2204`

There are new Variables
 * **VERSION** which contains whole version string
 * **LSBVERSION** which contains LSB version of distro
 * **LSBID** which contains LSB ID (Debian or Ubuntu)
added to `debian/autobake-deb.sh`.

Also **CODENAME** is change to **LSBNAME** as it's more declaritive
## How can this PR be tested?
This can be tested manually in Ubuntu or Debian
```
apt-get install git-buildpackage
mk-build-deps -r -i debian/control -t "apt-get -y -o Debug::pkgProblemResolver=yes --no-install-recommends"
debian/autobake-deb.sh
```
If compiling on Debian 11 after that there should be deb packages named: mariadb-*+maria~deb11.deb

This is also testes on 10.9 branch on Salsa-CI:
https://salsa.debian.org/illuusio/mariadb-server/-/pipelines/384410


## Basing the PR against the correct MariaDB version

## Backward compatibility
Makes sure that user can upgrade from Buster to Bookworm
